content/docs/user-guide/programming/components/reflection/reflecting-for-serialization.md

```diff
@@ -269,3 +271,63 @@ AZStd::vector<AZStd::string> MyComponent::GetEnabledOptions()
     return options;
 }
 ```
+
+It is also possible to bind the list of strings to values of another data type. One example is setting the component's fields to programmer-friendly values by using user-friendly names. To do so, set the `GenericValueList` attribute on the field's `DataElement` to a value that returns `AZStd::vector<AZStd::pair<T, AZStd::string>>`. The first parameter, `T`, in the `AZStd::pair` holds the programmer-friendly type and value, while the second holds the user-friendly string:
+
+```cpp
+...
+
+->Attribute(AZ::Edit::Attributes::GenericValueList, &MyComponent::GetEnabledOptions)
+
+//...
+AZStd::vector<AZStd::pair<AZ::Crc32, AZStd::string>> GetEnemyTypes()
+{
+    AZStd::vector<AZStd::pair<AZ::Crc32, AZStd::string>> options;
+    options.reserve(16);
+    options.push_back({AZ_CRC_CE("Basic"), AZStd::string("Basic option")});
+    options.push_back({AZ_CRC_CE("Basic2"), AZStd::string("Another basic option")});
+
+    if (m_enableAdvancedOptions)
+    {
+        options.push_back({AZ_CRC_CE("Advanced"), AZStd::string("Advanced option")});
+        options.push_back({AZ_CRC_CE("Advanced2"), AZStd::string("Another advanced option")});
+    }
+
+    return options;
+}
+```
+
+Within the O3DE editor, combo boxes work by default with the following field data types:
++ Numeric (`int`, `float`, etc.)
++ `AZStd::string`
++ Enums [reflected](#editing) to the `EditContext`. 
+
+To use the ComboBox with other types, register a combo box handler for the desired field type by calling `AzToolsFramework::RegisterGenericComboBoxHandler<T>()`, where `T` is the field type. Since the function lives in the `AzToolsFramework` namespace, make the call within a target that links to the Editor, such as a gem's `EditorSystemComponent::Activate` function.
```

Comment:
```suggestion
To use a combo box with other types, register a combo box handler for the desired field type by calling `AzToolsFramework::RegisterGenericComboBoxHandler<T>()`, where `T` is the field type. Since the function lives in the `AzToolsFramework` namespace, make the call within a target that links to the Editor, such as a Gem's `EditorSystemComponent::Activate` function.
```